backend/geometricFunctions.cpp

```diff
@@ -0,0 +1,123 @@
+#include "geometricFunctions.h"
+
+bool intersects (Line a, Line b) {
+    Point p = Point();
+    return intersects (a, b, p);
+}
+
+bool intersects (Line a, Line b, Point &p) {
+     // Line A
+    double a1 = a.getPoint(2).y() - a.getPoint(1).y();
```

Comment:
The variable names could maybe be more descriptive in this file

Reply:
_could maybe_
I would suggest not doing it as they are mostly helper variables not used for anything special.